tools/dtrace/nexus/trace-transactions.d

```diff
@@ -0,0 +1,59 @@
+#!/usr/sbin/dtrace -qs
+
+/* Trace all transactions to the control plane database with their latency */
+
+dtrace:::BEGIN
+{
+    printf("Tracing all database transactions for nexus PID %d, use Ctrl-C to exit\n", $target);
+}
+
+/*
+ * Record the start and number of statements for each transaction.
+ *
+ * We're only recording the initial transactions, which are not nested.
+ */
+diesel_db$target:::transaction-start
+/arg1 == 0/
+{
+    this->key = copyinstr(arg0);
+    ts[this->key] = timestamp;
+    n_statements[this->key] = 0;
+}
+
+/*
+ * When a query runs in the context of a transaction (on the same connection),
+ * bump the statement counter.
+ */
+diesel_db$target:::query-start
+/ts[copyinstr(arg1)]/
+{
+    n_statements[copyinstr(arg1)] += 1
+}
+
+/*
+ * As transactions complete, print the number of statements we ran and the
+ * duration.
+ *
+ * NOTE: The depth, arg1, is intentionally 1 here. When we first enter a
+ * non-nested transaction, the depth is 0, because there is no outer
+ * transaction. However, when the `transaction-done` probe fires, we are still
+ * in a transaction, in which case the depth is currently 1. It's decremented
+ * _after_ the transaction itself is committed or rolled back.
+ */
+diesel_db$targe:::transaction-done
```

Comment:
Should this be `diesel_db$target`, not `diesel_db$targe`?

Reply:
Yes! Good catch, thanks. I'm not sure how that crept in.